src/start/exceptions.md

```diff
@@ -1,7 +1,256 @@
 # Exceptions
 
-> **TODO** Cover statically registering exception handlers using the `exception`
-> attribute; the default exception handler (`DefaultHandler`); and safe local
-> `static mut` variables (non-reentrant functions). Also cover (perhaps in a
-> separate section?) the HardFault exception that handles crashes (e.g. SIGILL,
-> SIGSEGV).
+Exceptions, and interrupts, are a hardware mechanism by which the processor
+handles asynchronous events and fatal errors (e.g. executing an invalid
+instruction). Exceptions imply preemption and involve exception handlers,
+subroutines executed in response to the signal that triggered the event.
+
+The `cortex-m-rt` crate provides an [`exception`] attribute to declare exception
+handlers.
+
+[`exception`]: https://rust-embedded.github.io/cortex-m-rt/0.6.1/cortex_m_rt_macros/fn.exception.html
+
+``` rust
+// Exception handler for the SysTick (System Timer) exception
+#[exception]
+fn SysTick() {
+    // ..
+}
+```
+
+Other than the `exception` attribute exception handlers look like plain
+functions but there's one more difference: `exception` handlers can *not* be
+called by software. Following the previous example, the statement `SysTick();`
+would result in a compilation error.
+
+This behavior is pretty much intended and it's required to provide a feature:
+`static mut` variables declared *inside* `exception` handlers are *safe* to use.
+
+``` rust
+#[exception]
+fn SysTick() {
+    static mut COUNT: u32 = 0;
```

Comment:
Have you ever tried slotting something between a `static mut` declaration and the`fn` like a `use` statement? That a real fun exercise and head scratcher. ;)